packages/flutter/lib/src/painting/colors.dart

```diff
@@ -484,14 +483,14 @@ class ColorProperty extends DiagnosticsProperty<Color> {
        );
 
   @override
-  Map<String, Object> toJsonMap(DiagnosticsSerializationDelegate delegate) {
-    final Map<String, Object> json = super.toJsonMap(delegate);
+  Map<String, Object?> toJsonMap(DiagnosticsSerializationDelegate delegate) {
+    final Map<String, Object?> json = super.toJsonMap(delegate);
     if (value != null) {
       json['valueProperties'] = <String, Object>{
-        'red': value.red,
-        'green': value.green,
-        'blue': value.blue,
-        'alpha': value.alpha,
+        'red': value!.red,
```

Comment:
why do we need this? We just checked that value isn't null in this block

Reply:
because `value` is a getter/field a null check doesn't promote it to non-nullable.